daemon/src/maker_cfd.rs

```diff
@@ -439,9 +439,12 @@ where
         // state change. Once we know that we go for either an accept/reject scenario we
         // have to remove the current order.
         self.current_order_id = None;
-        self.takers
-            .send(maker_inc_connections::BroadcastOrder(None))
-            .await?;
+
+        // TODO: If I don't comment this out we hang forever!
```

Comment:
How could I have known that a deadlock could occur here? What should I be looking out for?

Reply:
1. Actors need a `&mut self` to invoke a handler
2. Sending a message invokes a handler

This is the important code-part in xtra: https://github.com/Restioson/xtra/blob/224004bc95dac12b8b8d2a14fea5ddf37eef6ae0/src/context.rs#L215-L249

If `.tick` never returns, it cannot read any more messages. But if a handler can only complete by returning from a `.send` call, it can never return. It would be unsound if two handlers were running concurrently because they both receive `&mut self`.